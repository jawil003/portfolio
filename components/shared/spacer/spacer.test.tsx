import { render } from "@testing-library/react";
import {
  Spacer,
  Props as SpacerProps,
} from "./spacer";

const variants: SpacerProps[] = [
  {},
  { width: "20px" },
  { height: "10px" },
];

describe("Spacer", () => {
  test.each(variants)(
    "Create Spacer with %p",
    (variant) => {
      const testSpacer = render(
        <Spacer {...variant} />,
      );

      expect(
        testSpacer.container.firstChild,
      ).toMatchSnapshot();
    },
  );
});
