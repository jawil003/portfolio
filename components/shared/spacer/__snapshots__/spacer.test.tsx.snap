// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer Create Spacer with {"height": "10px"} 1`] = `
<div
  style="height: 10px;"
/>
`;

exports[`Spacer Create Spacer with {"width": "20px"} 1`] = `
<div
  style="width: 20px;"
/>
`;

exports[`Spacer Create Spacer with {} 1`] = `<div />`;
